controllers: ION Discover DJ Pro

meltedpianoman opened the pull request:
As requested by https://github.com/mixxxdj/mixxx/pull/2893

I add the ION Discover DJ Pro hardware to the controller hardware section in the Mixxx manual.

meltedpianoman commented:
@Holzhaus : Can you please have a look?
Thanks!

Holzhaus commented:
Thanks. Here are some vector images of the Controller: [ion_discover_dj_pro.zip](https://github.com/mixxxdj/manual/files/5461429/ion_discover_dj_pro.zip)

Can you add numbered labels with Inkscape and then document the controls like on the [Pioneer DDJ-200](https://manual.mixxx.org/2.3/en/hardware/controllers/pioneer_ddj_200.html) page? The font for the labels is "Open Sans".

meltedpianoman commented:
Hi @Holzhaus , I updated the manual as you requested similar to the Pioneer DDJ-200 with a separate Deck & Mixer section.

Can you please have a look and give me your comments on this?

Thanks,
Ivo

Holzhaus commented:
Can you please open the SVG files in Inkscape and save them as "Optimized SVG" to reduce line count and file size?

Holzhaus commented on source/hardware/controllers/ion_discover_dj_pro.rst:
```diff
@@ -0,0 +1,94 @@
+Controls not yet working (or tested):
+
+- Library browsing
+- Cue control
```
What do mean by that? The controller does not have hotcue pads (judging from photos I found on Google)

Holzhaus replied:
Or are you referring to PFL?

meltedpianoman replied:
On 2nd thought: I compared the function of the Cue button on the controller with the Cue button in the Mixxx UI. And they seem to work similar (holding the Cue button plays the track from a fixed location... right?). So I guess the Cue button is actually working as it should be. It just didn't make sense to me... I'll update the documentation for that button as well.

meltedpianoman replied:
Done

Holzhaus commented on source/hardware/controllers/ion_discover_dj_pro.rst:
```diff
@@ -0,0 +1,94 @@
+
+Controls not yet working (or tested):
+
+- Library browsing
```
I think mapping the browse knob and the deck load buttons is a critical feature and shouldn't be hard to map). Could you take care of that?

meltedpianoman replied:
Welll... indeed the browse button + deck load buttons seems to work. So I can document that. I'll update the documentation on that one.

meltedpianoman replied:
Done

Holzhaus commented on source/hardware/controllers/ion_discover_dj_pro.rst:
```diff
@@ -0,0 +1,94 @@
+
+- Library browsing
+- Cue control
+- Autoloop
```
This should be easy to map. Check the `beatloop_X_activate` controls.

meltedpianoman replied:
See my remarks below... I really would like to finish this controller mapping and leave some features that are not yet working for the community to work on further... I hope you can live with that.

Holzhaus commented on source/hardware/controllers/ion_discover_dj_pro.rst:
```diff
@@ -0,0 +1,94 @@
+4    :hwlabel:`SCRATCH` button                                Toggle the scratch mode
+5    :hwlabel:`TEMPO` slider                                  Adjust the track playing speed
+6    :hwlabel:`PITCH BEND` buttons -/+                        Lower (-) or Raise (+) the pitch temporarily
+7    :hwlabel:`REVERSE` button                                Hold for reverse play
```
The button is labeled "Reverse Play" I think.

meltedpianoman replied:
Yes, you are right, I'll update this text.

meltedpianoman replied:
Done

Holzhaus reviewed (commented):
Thank you. 

What about the STUTTER buttons, the ALL MUTE buttons and the CUE VOLUME/MASTER VOLUME knobs?

These are currently not documented.

meltedpianoman commented:
> 
> 
> Thank you.
> 
> What about the STUTTER buttons, the ALL MUTE buttons and the CUE VOLUME/MASTER VOLUME knobs?
> 
> These are currently not documented.

Hi @Holzhaus, 

The CUE volume & Master volume knobs indeed control the audio of the two internal sound output channels, so I indeed will document that.

I understand the controller mapping is far from perfect now. What I mainly did was taking the intial effort from [joachim](https://mixxx.discourse.group/t/ion-discover-dj-pro-mapping-script/12120) and updated that one to the new XML scheme, inverted some sliders and fixed a filepath that did not work on a case sensitive file system like on an Linux machine.

I'm personally not a DJ, but I'm trying to get this controller supported in Mixxx so my son can use it (also with future versions of the Mixxx software). I was hoping that even though it is not complete and perfect, it at least is a first start to get this controller officially supported by Mixxx. Maybe other users that are missing certain functionality will make this controller map a little bit better in the future... My guess is that still some users are glad to have this controller mapping with at least some basic functionality, instead of no support for this controller at all.

The STUTTER & ALL MUTE buttons are currently not mapped at all. I propose that I list that in the "not yet working" section, together with the AUTOLOOP buttons. 

Is it acceptable by you to accept the current state of this controller mapping, even with its shortcomings? If yes: then I'll update the manual with your remarks and upload it for another review. If not: then I'm afraid that I'm going to give up on this Pull Request because I find it difficult to sufficient find time to contribute further on this. Don't get me wrong: I understand that you have to guard the quality of the mixxx software. So if the status of this controller is not acceptable, I'll can accept that. 

Regards,
Ivo

Holzhaus commented on source/hardware/controllers/ion_discover_dj_pro.rst:
```diff
@@ -0,0 +1,99 @@
+Known Issues
+------------
+
+Controls not yet working (or untested):
+
+- STUTTER
+- ALL MUTE
+- AUTOLOOP
```
```suggestion
Controls not yet working:

- :hwlabel:`STUTTER` button
- :hwlabel:`ALL MUTE` button
- :hwlabel:`AUTOLOOP` buttons
```

meltedpianoman replied:
Done

Holzhaus commented on source/hardware/controllers/ion_discover_dj_pro.rst:
```diff
@@ -0,0 +1,99 @@
+ION Discover DJ Pro
+===================
+
+.. figure:: ../../_static/controllers/ion_discover_dj_pro.jpg
+   :align: center
+   :width: 100%
+   :figwidth: 100%
+   :alt: ION Discover DJ Pro
+   :figclass: pretty-figures
+
+   ION Discover DJ Pro
+
```
We don't have permission to use this image and it's also low-res and doesn't add information. Please remove (and also the image file).

meltedpianoman replied:
Done

Holzhaus commented on source/hardware/controllers/ion_discover_dj_pro.rst:
```diff
@@ -0,0 +1,99 @@
+Controller Mapping
+------------------
+
+.. note:: Only controls are documented that are known to be working.
+
```
This should be removed, this is redundant with the "Known Issues" section.

```suggestion
```

meltedpianoman replied:
Done

Holzhaus reviewed (changes requested):
Sorry for taking so long to review. Already looks good, just two removals, then we can merge IMHO. Thanks you.

Holzhaus reviewed (approved):
LGTM, thank you!

Holzhaus commented on source/hardware/controllers/ion_discover_dj_pro.rst:
```diff
@@ -0,0 +1,88 @@
+
+-  `Mixxx User Forum <https://mixxx.discourse.group/t/ion-discover-dj-pro-mapping-script/12120>`__
+
+.. versionadded:: 2.2.5
```
Oops, there will be no 2.2.5, I'll change this to 2.3.

meltedpianoman replied:
Ok, thank @Holzhaus 

But.. what about the GitHub wiki?
Is that no longer maintained?
- I changed the version number also in the table on https://github.com/mixxxdj/mixxx/wiki/Hardware-Compatibility
- What about the https://github.com/mixxxdj/mixxx/wiki/Ion-Discover-DJ-Pro page?
